### PR TITLE
memory available and max-mem pod schedulable capacity checks

### PR DESCRIPTION
**Pull request by joelddiaz**

### CODE REVIEW

File: scripts/monitoring/cron-send-cluster-capacity.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+        if 'm' in num:
+            cpu_num = int(num.rstrip('m'))
+        else:
+            cpu_num = int(num) * 1000
+
+        return cpu_num
+
+    @staticmethod
+    def byte_me(num):
```
Comment (twiest) [inline scripts/monitoring/cron-send-cluster-capacity.py]: should be `convert_to_byte`.

Should also be in a library somewhere.

File: scripts/monitoring/cron-send-cluster-capacity.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+        print "Percentage of memory allocated: {}".format(memory_percentage)
+        print "Number of max-size nodes schedulable: {}".format(schedulable)
+
+        self.zagg_sender.add_zabbix_keys({zbx_key_mem_alloc:
+                                          int(memory_percentage)})
+        self.zagg_sender.add_zabbix_keys({zbx_key_max_pods: schedulable})
+
+    @staticmethod
+    def cpu_convert(num):
```
Comment (twiest) [inline scripts/monitoring/cron-send-cluster-capacity.py]: should be in a library somewhere.

File: scripts/monitoring/cron-send-cluster-capacity.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+    def run(self):
+        '''  Main function to run the check '''
+
+        self.parse_args()
+        self.zagg_sender = ZaggSender(verbose=self.args.verbose,
+                                      debug=self.args.debug)
+
+        master_cfg = []
+        with open('/etc/origin/master/master-config.yaml', 'r') as yml:
```
Comment (twiest) [inline scripts/monitoring/cron-send-cluster-capacity.py]: Should be able to pass this in. Should default to it, though.

NO MAGIC NUMBERS (or strings)

File: scripts/monitoring/cron-send-cluster-capacity.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+    def parse_args(self):
+        ''' parse the args from the cli '''
+
+        parser = argparse.ArgumentParser(description='Cluster capacity sender')
+        parser.add_argument('-v', '--verbose', action='store_true',
+                            default=None, help='Verbose?')
+        parser.add_argument('--debug', action='store_true',
+                            default=None, help='Debug?')
+        parser.add_argument('--test', action='store_true', default=False,
```
Comment (twiest) [inline scripts/monitoring/cron-send-cluster-capacity.py]: We've been using `--dry-run` for this. Would be nice to stay consistent.

File: scripts/monitoring/cron-send-cluster-capacity.py [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+                    pod['cpu_limits'] = pod.get('cpu_limits', 0) + \
+                                        self.cpu_convert(cpu)
+
+                mem = container['resources']['limits'].get('memory')
+                if mem:
+                    pod['memory_limits'] = pod.get('memory_limits', 0) + \
+                                           self.byte_me(mem)
+            else:
+                # no limit defined for container
+                pass
```
Comment (twiest) [inline scripts/monitoring/cron-send-cluster-capacity.py]: Why have an else clause at all?

File: scripts/monitoring/cron-send-cluster-capacity.py [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+                    pod['cpu_requests'] = pod.get('cpu_requests', 0) + \
+                                          self.cpu_convert(cpu)
+
+                mem = container['resources']['requests'].get('memory')
+                if mem:
+                    pod['memory_requests'] = pod.get('memory_requests', 0) + \
+                                             self.byte_me(mem)
+            else:
+                # no limit defined for container
+                pass
```
Comment (twiest) [inline scripts/monitoring/cron-send-cluster-capacity.py]: Why have an else clause at all?

**Comment by joelddiaz** [comment]
@twiest: split out the cpu and byte conversions into python-openshift-tools. you can now pass in your own master-node.yaml location. converted to --dry-run. removed useless 'else' clauses.

**Comment by twiest** [comment]
:+1:

**Pull request merged**